content/docs/projects/trust-manager/README.md

```diff
@@ -56,79 +76,293 @@ spec:
+also updating the trust-manager controller at the same as the default CA package:
+
 ```bash
-kubectl get cm -n kube-system example-bundle -o jsonpath="{.data['target-key']}"
-123
-ABC
-hello world!
+# Get the currently installed version. You could do this manually if you find that easier.
+TRUST_MANAGER_VER=$(helm list --filter "^trust-manager$" -n cert-manager -ojson | jq -r ".[0].app_version")
+
+# Check the version makes sense
+echo $TRUST_MANAGER_VER
+
+# Run the upgrade
+helm upgrade -f values.yaml -n cert-manager trust-manager jetstack/trust-manager --version $TRUST_MANAGER_VER
 ```
+
+## Preparing for Production
+
+TLS is a complicated beast and there are plenty of ways to use TLS certificates in ways which will cause problems
+down the line. We'd like to document a few of these potential gotchas here to help prepare users for running
+trust-manager safely in production.
+
+If you're planning on running trust-manager in production and you're using more than just the default CA package,
+we **strongly** advise you to read and understand this section. It could save you from causing an outage later.
+
+> ℹ️ These gotchas aren't specific to trust-manager and you could run into any of them with any method of managing TLS trust!
+
+### Bundling Intermediates
+
+If you've ever used a Let's Encrypt client such as [Certbot](https://certbot.eff.org/) you'll probably have
+seen that it generates several certificate files, such as `cert.pem`, `chain.pem`, and `fullchain.pem`.
+
+These various files are provided to support various different applications, which might require the certificate
+and the chain to be given separately. For most users and applications `fullchain.pem` is the only correct choice.
+
+Unfortunately the existence of these files has the unfortunate side effect of people sometimes assuming that `cert.pem`
+is the correct choice even when `fullchain.pem` would be correct. This means that the rest of the chain will not
+be sent when the certificate is used.
+
+Often, a quick fix that _seems_ to work for this is that clients add the chain to their trust store, which will seem
+to fix certificate errors in the short term. It's easy for this kind of "fix" to end up being embedded somewhere as a
+solution which others can follow.
+
+This "fix" is dangerous; it means that the intermediate cannot be safely rotated without all trust stores
+which contain it being updated first.
+
+Intermediates in this case become _de facto_ root certificates, which completely defeats the point of having
+intermediate certificates in the first place.
+
+Avoid using intermediates in any trust store wherever possible unless you're absolutely certain they should be included.
+An example of where it might be OK would be cross signing, which is not likely to be required in the general case.
+
+It would be better to copy just the root certificate to a new `ConfigMap` and use that as a source rather than trusting
+an intermediate.
```

Comment:
I find this section confusing because 

> For most users and applications `fullchain.pem` is the only correct choice.

contradicts

> Avoid using intermediates in any trust store wherever possible unless you're absolutely certain they should be included.

If the section is meant to be about what file users should instruct trust-manager to copy to the trust store.

Given the CA certificates:
 *  `Example Corp Root > Cloud Infrastructure > Linkerd`
 *  `Example Corp Root > VPN > Sales`

I would not want my linkerd client to trust a peer that presented a certificate signed by `Example Corp Root > VPN > Sales`.
So I would configure trust-manager to copy the intermediate CA  `Example Corp Root > Cloud Infrastructure > Linkerd` into the ConfigMap `linkerd/linkerd-identity-trust-roots`.

Reply:
> I find this section confusing because ...

That makes sense, yeah. I can see how that would be confusing.

>  So I would configure trust-manager to copy the intermediate CA Example Corp Root > Cloud Infrastructure > Linkerd into the ConfigMap linkerd/linkerd-identity-trust-roots.

This kinda thing feels like an antipattern to me. It's _so_ easy to get this wrong and it turns every single intermediate into a de facto root certificate. It would be much better to have two roots, "Corp Root Infra" and "Corp Root VPN", in this scenario.

I think orgs would be well advised not to create PKI structures like this if at all possible. Certainly I wouldn't actively recommend it, although I acknowledge that this pattern exists today.